packages/dev/core/src/Engines/nativeEngine.ts

```diff
@@ -2704,4 +2717,18 @@ export class NativeEngine extends Engine {
                 return buffer;
             });
     }
+
+    override startTimeQuery(): Nullable<_TimeToken> {
+        if (!this._gpuFrameTimeToken) {
+            this._gpuFrameTimeToken = new _TimeToken();
+        }
+
+        // Always return the same time token. For native, we don't need a start marker, we just query for native frame stats.
+        return this._gpuFrameTimeToken;
+    }
+
+    override endTimeQuery(token: _TimeToken): int {
+        this._engine.populateFrameStats(this._frameStats);

```

Comment:
@bghgary how are we handling versioning these days? Do we use `PROTOCOL_VERSION` for a change like this, or fail gracefully if the `populateFrameStats` function doesn't exist?

Reply:
I would do the latter. Make `populateFrameStats` optional in the interface.